service/src/PlatformIO.cpp

```diff
@@ -105,31 +102,61 @@ namespace geopm
         m_iogroup_list.push_back(iogroup);
     }
 
-    std::shared_ptr<IOGroup> PlatformIOImp::find_signal_iogroup(const std::string &signal_name) const
+    std::vector<std::shared_ptr<IOGroup> > PlatformIOImp::find_signal_iogroup(const std::string &signal_name) const
     {
-        std::shared_ptr<IOGroup> result = nullptr;
-        bool is_found = false;
+        std::vector<std::shared_ptr<IOGroup> > result;
+        int native_domain = GEOPM_DOMAIN_INVALID;
         for (auto it = m_iogroup_list.rbegin();
-             !is_found && it != m_iogroup_list.rend();
+             it != m_iogroup_list.rend();
              ++it) {
             if ((*it)->is_valid_signal(signal_name)) {
-                result = *it;
-                is_found = true;
+                if (result.empty()) {
+                    result.push_back(*it);
+                    native_domain = (*it)->signal_domain_type(signal_name);
+                }
+                else if ((*it)->signal_domain_type(signal_name) == native_domain) {
+                    result.push_back(*it);
+                }
+#ifdef GEOPM_DEBUG
+                else {
+                    std::cerr << "Warning: <geopm> PlatformIO::find_signal_iogroup(): "
```

Comment:
Native domain differs for the same provided by another IOGroup.
The current implementation does not support fallback using an IOGroup
that provides a signal with a different native domain.  If this is an issue
for your use case, please report to the developers requesting this feature.
The signal << signal_name << will not use << (*it)->name() for fallback
if a read fails.